### PR TITLE
Support pagingIo in mirror.net

### PR DESCRIPTION
**Pull request by TrabacchinLuigi**
Little refactor to remove some warnings and tidy up

### CODE REVIEW

**Comment by TrabacchinLuigi** [comment]
Should fix #294

**Comment by TrabacchinLuigi** [comment]
also : is really an uint in unsafe mirror ? can really be that long? i mean, it's still 32bit, and makes more sense, but everywhere else an int32 is used

File: sample/DokanNetMirror/NativeMethods.cs [hunk 1]
```diff
@@ -25,5 +27,14 @@ public static class NativeMethods
         [DllImport("kernel32", SetLastError = true)]
         [return: MarshalAs(UnmanagedType.Bool)]
         public static extern bool SetFileTime(SafeFileHandle hFile, ref long lpCreationTime, ref long lpLastAccessTime, ref long lpLastWriteTime);
+        [DllImport("kernel32.dll", SetLastError = true)]
```
Comment (Liryna) [inline sample/DokanNetMirror/NativeMethods.cs:30]: return line missing

**Review by Liryna — CHANGES_REQUESTED** [review]

**Comment by Liryna** [comment]
Thanks @TrabacchinLuigi for fixing the issue! Very appreciated 😎 

Just requested a small change before merge if thats not a problem

**Comment by TrabacchinLuigi** [comment]
i've lot of changes right now, if it's not a problem i'll change that on next pull request

**Comment by TrabacchinLuigi** [comment]
also, i was thinking some atributes are really not needed like marshaling boolean to boolean is the default choiche so no need to explicitly specify that

**Comment by TrabacchinLuigi** [comment]
Dang since i've not made a feture branch i can't do another pull request, git is not my bread and butter, TFS is so much easier

**Comment by Liryna** [comment]
Yeah when you work on the same branch due to your second commit based on another one that is also a PR, everything ends up in the same PR. That's fine if they are clean commit like yours 👍 
Thanks for taking care of the issue your reported!

**Pull request merged**